src/makeInfinityAdder.js

```diff
@@ -4,7 +4,24 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  // eslint-disable-next-line no-unused-vars
+  let sum = 0;
+
+  const sumNumbers = (arg) => {
+    if (arg === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += arg;
+
+    return sumNumbers;
```

Comment:
The function `sumNumbers` should return itself to allow chaining, not the updated sum. Returning the sum here would cause the next chained call to fail, as numbers can't be called as functions.